doc/building-and-running.md

```diff
@@ -0,0 +1,76 @@
+MAGDA can be run in two main ways:
+
+* Kubernetes - Each component is built into a Docker container and run on a Kubernetes cluster, which may be a robust production environment or a simple local cluster like Minikube.  This is good for production environments and for testing in a more production-like environment on your development machine.  It also creates a more consistent environment, potentially avoiding configuration quirks of your development machine.
+* Host - The components are run in JVM processes running directly on your development machine.  This allows for fast development cycles because a code change can quickly trigger a restart of the JVM running that component. 
+
+## Kubernetes (local, minikube)
+
+Running on a local Kubernetes cluster involves these steps:
+
+1. Set up your cluster.  On a development system, this means installing minikube and kubectl.
+2. Deploy the kube registry to the cluster.
+
+```bash
+kubectl create -f deploy/kubernetes/registry.yml
+```
+
+3. Compile the components, create their Docker images, and push them to the local registry.  This will also terminate the running pods associated with the components, triggering them to restart with the new image.
+
+```bash
+sbt deployLocal
+```
+
+4. Deploy to the cluster
+ 
+```bash
+cd deploy/kubernetes
+kubectl create -f elasticsearch.yml -f registry-datastore.yml -f local.yml
+```
+
+At this point, the services will be running on your Minikube IP address.  Use `minikube docker-env` and take a look at the IP address listed under `DOCKER_HOST`.  The port of each service will be the `nodePort` listed in `local.yml`.
+
+In development, it may be helpful to run:
+
+```bash
+sbt ~deployLocalOnChange
+```
+
+This will automatically re-run `deployLocal` for a component each time a source file used by that component changes. 
+
+## Host
+
+Running the JVM-based components directly on your development machine is easy.  However, a working MAGDA system also needs these components:
+
```

Comment:
Should probably make sure we mention that you need to follow the start of the ## Kubernetes section to get this to work